server/src/main/java/org/elasticsearch/search/fetch/subphase/ScriptFieldsPhase.java

```diff
@@ -22,10 +22,19 @@
 import java.util.List;
 
 public final class ScriptFieldsPhase implements FetchSubPhase {
+    @Override
+    public String name() {
+        return "script_fields";
+    }
+
+    @Override
+    public String description() {
+        return "run scripts and return their results";
+    }
 
     @Override
     public FetchSubPhaseProcessor getProcessor(FetchContext context) {
-        if (context.scriptFields() == null) {
+        if (context.scriptFields() == null || context.scriptFields().fields().isEmpty()) {
```

Comment:
When you don't ask for any script fields we tend to get an empty list here, not null. That empty list was causing us to always return the phase and run it. That's not a big deal at all because it just iterates an empty list, but it was clogging up the profile output.